### DISCUSSION TITLE
MicrobiomeStat_v1.1.3

### DISCUSSION THREAD

**Discussion by cafferychen777** [cafferychen777/MicrobiomeStat#21] [Announcements]
[MicrobiomeStat_1.1.3.tar.gz](https://github.com/cafferychen777/MicrobiomeStat/files/13578964/MicrobiomeStat_1.1.3.tar.gz)

<hr /><em>This discussion was created from the release <a href='https://github.com/cafferychen777/MicrobiomeStat/releases/tag/MicrobiomeStat_Version_1.1.3'>MicrobiomeStat_v1.1.3</a>.</em>

**Comment by 16svale** [comment]
Great that there is an updated version of the MicrobiomeStat!
I am wondering if there is a way to make sure that the labels of ,for example the "Family", do not overlap in the volcano plots.
I am struggling to visualize the entire name like in this screenshot.
I hope you can help.
Best regards,
Valentina

![Capture](https://github.com/cafferychen777/MicrobiomeStat/assets/93317465/d046199a-c57d-4f26-9497-f43444ee2643)

**Comment by cafferychen777** [comment]
Dear @16svale,

Thank you for reaching out and for your interest in the updated version of MicrobiomeStat!

I'm glad to inform you that I have recently made some modifications to the text display in the volcano plots to address the issue of label overlap. In the latest update pushed to GitHub, the labels, such as those for "Family," should no longer overlap, making it easier to visualize the entire name.

Please try updating the MicrobiomeStat package tomorrow and check if the changes meet your needs. If you encounter any further issues or have additional feedback, feel free to reach out.

Best regards,
Chen YANG

**Reply by 16svale** [reply]
@cafferychen777 Thank you very much for your quick response.
I actually installed the MicrobiomeStat for the first time on Friday 16 February 2024 and I see that I have the latest version 1.1.3 and still have the overlapping issue. Is there something else I need to update? If yes, could you attach the link or the explanation on how to update it? Thanks
![image](https://github.com/cafferychen777/MicrobiomeStat/assets/93317465/746c4b2d-e238-4f39-a9be-2a0a9e206af1)

**Reply by cafferychen777** [reply]
Dear Valentina,

Thank you for getting back to me, and I appreciate your patience.

It seems there might have been a misunderstanding. While I've updated the function to address the label overlap issue, the version number of the R package itself remains the same (1.1.3). Therefore, you won't receive the updated functionality unless you reinstall the package from GitHub.

To ensure you have the latest changes, please follow these steps:

1. Install the devtools package if you haven't already:
   ```R
   install.packages("devtools")
   ```

2. Reinstall the MicrobiomeStat package from GitHub:
   ```R
   devtools::install_github("cafferychen777/MicrobiomeStat")
   ```

After reinstalling the package using the provided command, you should have access to the updated functionality without the label overlap issue. Please give it a try, and let me know if you encounter any further difficulties.

Thank you for your understanding and cooperation.

Best regards,
Chen YANG

**Reply by 16svale** [reply]
Oh Great, now it works! Thank you for answering. This is a great package.

I am wondering how I can change the reference level. For example my variable is called sampletype and I have three levels "sand", water_E" and water_I".  sand is used as a reference because of the alphabetic order and it is ok for part of the analysis. But I also want to compare only waterE and water_I. How can I do it? 

Best regard,
Valentina

**Reply by cafferychen777** [reply]
Dear Valentina,

I'm glad to hear that the package is working well for you!

Regarding your question about changing the reference level for the variable "sampletype" when running linda to compare only "water_E" and "water_I," you can adjust the factor levels within the "meta.dat" of the "data.obj" object.

Here's how you can achieve this:

```R
# Assuming data.obj is your object containing the data
# Check the current levels of the sampletype variable
levels(data.obj$meta.dat$sampletype)

# Change the reference level to "water_E"
data.obj$meta.dat$sampletype <- factor(data.obj$meta.dat$sampletype, levels = c("water_E", "water_I", "sand"))

# Now, "water_E" will be the reference level for comparisons
```

By setting the levels of the "sampletype" variable in the metadata of your data object, you can control which level serves as the reference for your analyses.

Feel free to try this approach, and let me know if you have any further questions or need additional assistance.

Best regards,
Chen Yang

**Reply by 16svale** [reply]
Thank for you answer.

Are you also planning to include the possibility to create plot similar to the one in this picture in addition to volcano plots? I sit alredy possible?
I think it would be a valuable addition to the package because a lot of differential abundance results are presented in this way.
![image](https://github.com/cafferychen777/MicrobiomeStat/assets/93317465/b50dffe7-3721-434e-b1e0-50b52b7fc255)

Thanks

**Reply by cafferychen777** [reply]
Dear Valentina,

Thank you for your inquiry.

While I appreciate your suggestion regarding the possibility of including plot types similar to the one in the provided image in addition to volcano plots, from a scientific standpoint, the information presented in both types of plots is essentially the same. From an information theory perspective, they convey identical information.

Therefore, at this time, I do not plan to include this specific plot format in the package. However, I am continuously evaluating and considering feedback from users like you to enhance the package's functionality and usability.

If you have any further questions or suggestions, please don't hesitate to reach out. Your input is valuable and helps improve the package for all users.

Best regards,
Chen Yang

**Reply by 16svale** [reply]
Thank you for all your answers!
